pkg/cmd/login/login.go

```diff
@@ -103,7 +104,20 @@ func NewLoginCmd(f *factory.Factory) *cobra.Command {
 				opts.Logger.Info(opts.localizer.MustLocalize("login.log.info.sshLoginDetected", localize.NewEntry("OfflineTokenURL", build.OfflineTokenURL)))
 			}
 
-			return runLogin(opts)
+			ch := make(chan error, 1)
```

Comment:
My take would be to have timeout util rather than code directly embedded into login function. 

Timeout can be done many different ways. This  puts timeout on entire login method. 
We can also do it individually using middleware: https://github.com/go-http-utils/timeout

My take is that this approach is better although it needs to be done as separate function/helper

Reply:
Just checked. We cannot use http middleware in this case. This code is the best way forward IMHO